client/lib/cart/store/index.js

```diff
@@ -118,8 +111,8 @@ CartStore.dispatchToken = Dispatcher.register( function( payload ) {
 			update( cartItems.removePrivacyFromAllDomains( CartStore.get() ) );
 			break;
 
-		case UpgradesActionTypes.CART_ITEM_ADD:
-			update( cartItems.add( cartItem ) );
+		case UpgradesActionTypes.CART_ITEMS_ADD:
+			update( flow( ...action.cartItems.map( cartItem => cartItems.add( cartItem ) ) ) );
```

Comment:
Mine, too :)

I had to use some bits of functional programming combined with ES6 to get this combined into one server update.

If you have any idea how to refactor it, let me know :)